support/cas-server-support-pm/src/main/java/org/apereo/cas/pm/PasswordManagementService.java

```diff
@@ -63,4 +63,20 @@ default String parseToken(String token) {
     default Map<String, String> getSecurityQuestions(String username) {
         return new LinkedHashMap<>();
     }
+
+    /**
+     * Checks a security questions answer.
+     *
+     * @param username the username
+     * @param question the text of the question
+     * @param answer stored answer
+     * @param input user response to question
+     * @return whether the answer is correct
+     */
+    default boolean checkSecurityQuestionAnswer(String username, String question, String answer, String input) {
```

Comment:
While this is fine, I suspect we might want to refactor this in the future to expose a "validator" interface of some kind where you'd write the implementation for that only to validate answers.